open_tms_backend/open_tms_create_ticket/index.js

```diff
@@ -1,19 +1,143 @@
 import { shared } from '@appblocks/node-sdk'
 
-const handler = async (event) => {
-  const { req, res } = event
-  const { sendResponse } = await shared.getShared()
+const allRequiredSet = (scope, ...properties) =>
+  scope && (!properties || properties.every(key => !key || key in scope))
 
-  // health check
-  if (req.params.health === 'health') {
-    return sendResponse(res, 200, { message: 'Health check success' })
+async function getStageIdByName(stageName) {
+  const { prisma } = await shared.getShared()
+
+  // Retrieves the unique identifier (ID) associated with a specific stage name from a database table
+  const stage = await prisma.stage.findUnique({
+    where: { name: stageName },
+  });
+  return stage.id;
+}
+
+async function getOrganizationId(userID) {
+  const { prisma } = await shared.getShared()
+
+  const queryResult = await prisma.$queryRaw`
+        SELECT om.organisation_id
+        FROM org_member_roles AS omr
+        JOIN org_member AS om ON omr.user_id = om.created_by
+        WHERE omr.user_id = ${userID};
+      `;
+
+  if (queryResult[0]) {
+    const organisation_id = queryResult[0].organisation_id;
+    return organisation_id;
+  } else {
+    console.error('User not found in org_member_roles.');
+    return null
+  }
+}
+
+// const handler = async (event) => {
+const handler = async (req) => {
+  const { sendResponse, prisma } = await shared.getShared()
+
+  // const { req, res } = event
+
+  try {
+
+    // Perform Health check
+    // if (req.params.health === 'health') {
+    //   console.log('Health check success')
+    //   // sendResponse(res, 400, { message: 'Health check success' })
+    // }
+
+    // User context from authentication middleware
+    const creator_id = req?.user?.id
+    if (!creator_id) {
+      console.log("Unauthorized!")
+      // return sendResponse(res, 401, { message: "Unauthorized!" })
+    }
+    const organisation_id = getOrganizationId(creator_id)
+
+
+    if (req?.body && allRequiredSet(req?.body, "name", "department", "description", "ticket_type")) {
+
+      return await prisma.$transaction(async (tx) => {
+
+        // Create an entry in the tickets table
+        const ticket = await prisma.ticket.create({
+          data: {
+            created_by: creator_id,
+            organisation_id
+          },
+        });
+
```

Comment:
Please create ticket_revision and ticket_activity with the correct assignee_id for the given ticket.Add these two as part of this transaction.